builds/ngap/beslog2json.py

```diff
@@ -0,0 +1,560 @@
+# -*- mode: python; c-basic-offset:4 -*-
+#
+# This file is part of the Hyrax Docker Project
+#
+# Copyright (c) 2024 OPeNDAP, Inc.
+# Author: Nathan Potter <ndp@opendap.org>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# You can contact OPeNDAP, Inc. at PO Box 112, Saunderstown, RI. 02874-0112.
+#
+# ----------------------------------------------------------------------------------
+# Program beslog2json.py
+# Converts BES log lines with the formats described below in to
+# a json record.
+#
+# Log Fields type==request
+# 1601646465|&|2122|&|request|&|OLFS|&|0:0:0:0:0:0:0:1|&|USER_AGENT|&|92F3C71F959B56515C98A09088CA2A8E
+#    1          2        3       4           5              6               7
+# |&|-|&|1601646465304|&|REQUEST-ID|&|HTTP-GET|&|/opendap/hyrax/data/nc/fnoc1.nc.dds|&|u|&|BES
+#    8     9             10             11              12                             13   14
+# |&|get.dds|&|dap2|&|/Users/ndp/OPeNDAP/hyrax/build/share/hyrax/data/nc/fnoc1.nc|&|u
+#      15       16                              17                                  18
+#
+# 1601642679|&|2122|&|request|&|RequestFields
+# 1601642669|&|2122|&|timing|&|TimingFields
+# 1601642669|&|2122|&|verbose|&|MessageField
+# 1601642679|&|2122|&|info|&|MessageField
+# 1601642679|&|2122|&|error|&|MessageField
+#
+import sys
+import re
+import json
+import getopt
+import ast
+import time
+import os
+
+
+##########################################################
+# Global variable for debuggin control.
+debug_flag = False 
+
+##########################################################
+bes_log_field_delimiter = "|&|"
+bes_square_bracket_log_delimiter = "]["
+
+##########################################################
+# debug()
+# Writes "msg" to stderr when the debug_flag is enabled
```

Comment:
suggestion: include the definition of the function debug() within the scope of the function. For example:
```python
def debug(msg):
    """ writes `msg` to stderr when debug_flag is enabled
    """
```

Reply:
but... How is this useful to make the descriptions "data" that cannot be accessed? What is this?

Reply:
Are the descriptions (docstring) to debug()  (```Writes "msg" to stderr when the debug_flag is enabled```) being accessed by another function/script? My comment/suggestion has to do with a standardized way to define a function via [their docstrings](https://peps.python.org/pep-0257/)